ts/screens/profile/ServicesPreferenceScreen.tsx

```diff
@@ -1,75 +1,110 @@
  * User can update his/her mode
  * @param props
  * @constructor
  */
-const ServicesPreferenceScreen = (props: Props): React.ReactElement => {
-  const { potProfile, state } = props;
+const ServicesPreferenceScreen = (): ReactElement => {
+  const store = useStore();
+  const state = store.getState();
+  const dispatch = useIODispatch();
+  const profile = useIOSelector(profileSelector);
+  const prevProfile = usePrevious(profile);
+  const isLoading = pot.isUpdating(profile) || pot.isLoading(profile);
+  const profileServicePreferenceMode = useIOSelector(
+    profileServicePreferencesModeSelector
+  );
+
+  const dispatchServicePreferencesSetting = useCallback(
+    (mode: ServicesPreferencesModeEnum) =>
+      dispatch(
+        profileUpsert.request({ service_preferences_settings: { mode } })
+      ),
+    [dispatch]
+  );
+
+  const onServicePreferenceSelected = (
+    mode: ServicesPreferencesModeEnum,
+    state: GlobalState
+  ) => {
+    void trackServiceConfiguration(mode, getFlowType(false, false), state);
+    dispatchServicePreferencesSetting(mode);
+  };
+
   const { present: confirmManualConfig, manualConfigBottomSheet } =
     useManualConfigBottomSheet(() =>
-      props.onServicePreferenceSelected(
-        ServicesPreferencesModeEnum.MANUAL,
-        state
-      )
+      onServicePreferenceSelected(ServicesPreferencesModeEnum.MANUAL, state)
     );
 
   useOnFirstRender(() => {
     trackServiceConfigurationScreen(getFlowType(false, false));
   });
 
-  const [prevPotProfile, setPrevPotProfile] = React.useState<
-    typeof props.potProfile
-  >(props.potProfile);
-  React.useEffect(() => {
+  useEffect(() => {
     // show error toast only when the profile updating fails
-    // otherwise, if the profile is in error state, the toast will be shown immediately without any updates
-    if (!pot.isError(prevPotProfile) && pot.isError(potProfile)) {
-      showToast(I18n.t("global.genericError"));
+    // otherwise, if the profile is in error state,
+    // the toast will be shown immediately without any updates
+    if (
+      prevProfile !== undefined &&
+      !pot.isError(prevProfile) &&
+      pot.isError(profile)
+    ) {
+      IOToast.error(I18n.t("global.genericError"));
+      return;
     }
-
-    setPrevPotProfile(potProfile);
-  }, [potProfile, prevPotProfile]);
+    // if profile preferences are updated correctly
+    // the button is selected
+    // and the success banner is shown
+    if (
+      prevProfile !== undefined &&
+      pot.isUpdating(prevProfile) &&
+      pot.isSome(profile)
+    ) {
+      IOToast.success(
+        profileServicePreferenceMode === ServicesPreferencesModeEnum.MANUAL
+          ? I18n.t("services.optIn.preferences.manualConfig.successAlert")
+          : I18n.t("services.optIn.preferences.quickConfig.successAlert")
+      );
+    }
+  }, [profile, prevProfile, profileServicePreferenceMode]);
 
   const handleOnSelectMode = (mode: ServicesPreferencesModeEnum) => {
```

Comment:
What about using a callback? What do you think?

Reply:
done!